check-diff.sh

```diff
@@ -624,6 +624,23 @@ function lint_js_files {
 	fi
 }
 
+function lint_xml_files {
+	if [ ! -s "$TEMP_DIRECTORY/paths-scope-xml" ]; then
+		return
+	fi
+
+	set -e
+
+	echo "## XMLLINT"
+	cd "$LINTING_DIRECTORY"
+	if ! cat "$TEMP_DIRECTORY/paths-scope-xml" | remove_diff_range | xargs xmllint --noout 2> "$TEMP_DIRECTORY/xmllint-report"; then
+		if [ -s "$TEMP_DIRECTORY/xmllint-report" ]; then
+			cat "$TEMP_DIRECTORY/xmllint-report"
+			exit 1
+		fi
+	fi
```

Comment:
@marcin-lawrowski I wonder, can this condition be simplified to:

```bash
cat "$TEMP_DIRECTORY/paths-scope-xml" | remove_diff_range | xargs xmllint --noout
```

Since we don't need to look at patch subsets of XML files (since one syntax error causes complete file to be unreadable), I think we can eliminate the need for `xmllint-report`